### PR TITLE
Error messages for `rename_columns` and `Vector.duplicates`

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Improve error message for `rename_columns`.
- Add `length` to `Set` and `Map`.
- Add `duplicates` to `Vector` (and `Array`).

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso [hunk 1]
```diff
@@ -47,12 +47,17 @@ type Set
...
-    ## GROUP Metadata
-       ICON metadata
+    ## ICON metadata
        Returns the number of elements in this set.
     size : Integer
     size self = self.underlying_map.size
 
+    ## GROUP Metadata
+       ICON metadata
+       Returns the number of entries in this set.
```
Comment (AdRiley) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso]: Can we make the descriptions the same? I had to think if there was a difference between elements and entries

Comment (jdunkerley) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Set.enso]: Done.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso [hunk 2]
```diff
@@ -311,10 +311,17 @@ rename_columns (naming_helper : Column_Naming_Helper) (internal_columns:Vector)
...
-            if map.is_error then Error.throw (Illegal_Argument.Error "A mapping Vector must be either a list of names or a list of pairs (old name to new name).") else
-                rename_columns naming_helper internal_columns map case_sensitivity error_on_missing_columns on_problems
+            ## Check all pairs (Integer | Text | Regex => Text )
+            is_valid_row r = r.is_a Vector || r.is_a Pair
+            is_valid_key k = k.is_a Integer || k.is_a Text || k.is_a Regex
+            all_pairs = mapping.all p-> (is_valid_row p) && p.length == 2 && (is_valid_key p.first) && p.second.is_a Text
+            if all_pairs.not then Error.throw (Illegal_Argument.Error "mapping is not a Vector of old name to new name.") else
+                ## Attempt to treat as Map
+                map = Map.from_vector mapping error_on_duplicates=False
+                if map.size == mapping.length then rename_columns naming_helper internal_columns map case_sensitivity error_on_missing_columns on_problems else
```
Comment (AdRiley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso]:
```suggestion
                if map.length == mapping.length then rename_columns naming_helper internal_columns map case_sensitivity error_on_missing_columns on_problems else
```

Comment (AdRiley) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso]: nit, but I think easier to read

File: distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso [hunk 3]
```diff
@@ -311,10 +311,17 @@ rename_columns (naming_helper : Column_Naming_Helper) (internal_columns:Vector)
...
+                    duplicates = mapping.duplicates on=_.first . map .to_text
+                    duplicate_text = duplicates.take 3 . to_vector . join ", " + (if duplicates.length > 3 then ", ..." + (duplicates.length - 3).to_text + " others" else "")
+                    Error.throw (Illegal_Argument.Error "duplicate old name mappings ("+duplicate_text+").")
```
Comment (AdRiley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Internal/Table_Helpers.enso:325]: Can we add tests for this?

**Review by AdRiley — APPROVED** [review]
Looks good! A much better error message for the end users :)

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**